pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -219,11 +224,16 @@ protected void sendMessagesToConsumers(ReadType readType, List<Entry> entries) {
                         sendMessageInfo.getTotalMessages(),
                         sendMessageInfo.getTotalBytes(), sendMessageInfo.getTotalChunkedMessages(),
                         getRedeliveryTracker()).addListener(future -> {
-                    if (future.isDone() && keyNumbers.decrementAndGet() == 0) {
-                        readMoreEntries();
-                    }
-                });
-
+                            if (future.isSuccess()) {
+                                Entry entry = entriesWithSameKey.get(entriesWithSameKey.size() - 1);
+                                Integer hashCode =
+                                        selector.generateKeyHash(peekStickyKey(entry.getDataBuffer()));
+                                currentSendPositionPerKeyMap.put(hashCode, entry.getPosition());
```

Comment:
This can potentially be storing a very large amount of `PositionImpl` instances, that will get tenured and will put pressure on the GC.

Reply:
Also, this would break if there is a hash collision.

Reply:
Same hash will use same consumer，so there is not problem when there is a hash collision, this map only record one position to one hash key,

Reply:
Now i change the code for this question.